### PR TITLE
Clone dialog - Dont disable filter while loading list of Repositories

### PR DESCRIPTION
**Pull request by danwalmsley**
Hello all,

This PR addresses issues #108 and #517. I also added a few unit tests to cover this scenario.

:)

### CODE REVIEW

**Comment by danwalmsley** [comment]
Looks like one of my unit tests failed. I will check why and update.

**Comment by grokys** [comment]
Yeah, it's failing here locally as well.

**Comment by danwalmsley** [comment]
@grokys I corrected logic and fixed unit tests now :)

File: src/GitHub.App/ViewModels/RepositoryCloneViewModel.cs [hunk 1]
```diff
@@ -75,12 +75,12 @@ public class RepositoryCloneViewModel : BaseViewModel, IRepositoryCloneViewModel
             repositories.Filter = FilterRepository;
             repositories.NewerComparer = OrderedComparer<IRemoteRepositoryModel>.OrderByDescending(x => x.UpdatedAt).Compare;
 
-            filterTextIsEnabled = this.WhenAny(x => x.IsLoading, x => x.Value)
-                .Select(x => !x && repositories.UnfilteredCount > 0)
+            filterTextIsEnabled = this.WhenAny(x => x.IsLoading,
+                (loading) => loading.Value || repositories.UnfilteredCount > 0 && !LoadingFailed)
```
Comment (grokys) [inline src/GitHub.App/ViewModels/RepositoryCloneViewModel.cs]: Tiny nit: we usually don't put brackets around a single lambda parameter.

**Comment by grokys** [comment]
Thanks @danwalmsley LGTM! If you could merge the latest changes from master in, I can merge.

File: src/GitHub.App/ViewModels/RepositoryCloneViewModel.cs (hunk 1 above)
Comment (danwalmsley) [inline src/GitHub.App/ViewModels/RepositoryCloneViewModel.cs]: Let me correct this before you merge :)

Comment (danwalmsley) [inline src/GitHub.App/ViewModels/RepositoryCloneViewModel.cs]: ok fixed this.

**Comment by danwalmsley** [comment]
Should be ready if you guys are happy :)

**Comment by grokys** [comment]
Merged 🎉 - thanks Dan!

**Pull request merged**